### ISSUE TITLE
PutObject does not respect embedded/anonymous fields

### ISSUE THREAD

**Issue by on99** [aerospike/aerospike-client-go#286]
I create a test file in aerospike-client-go package

```go
package aerospike

import "testing"

type record struct {
	a
	b
}

type a struct {
	A int `as:"a"`
}

type b struct {
	B string `as:"b"`
}

func TestMarshal(t *testing.T) {
	var r record
	r.A = 10
	r.B = "b"
	v := marshal(r, true)
	t.Log(v)

	// expected
	t.Log(map[string]interface{}{
		"a": 10,
		"b": "b",
	})
}
```

`go test -v -run Marshal` outputs `map[]` while the expected output should be `map[a:10 b:b]`

 https://github.com/jmoiron/sqlx/blob/master/reflectx/reflect_test.go#L38-L94 The implementation by sqlx is very elegant.

**Comment by on99** [comment]
refs #180

**Comment by khaf** [comment]
Released in v2.9.0.
